src/api/mergedData.js

```diff
@@ -24,12 +24,12 @@ const viewAuthorDetails = (authorFirebaseKey) => new Promise((resolve, reject) =
 });
 
 // We want to WAIT to for ALL books before we delete them.
-const deleteAuthorsBooks = (authorId) => new Promise((resolve, reject) => {
+const deleteAuthorsBooks = (authorId, uid) => new Promise((resolve, reject) => {
   getAuthorBooks(authorId).then((booksArray) => {
     const deleteBookPromises = booksArray.map((book) => deleteBook(book.firebaseKey));
     // Promise.all is expecting and ARRAY of Promises ^^^ from the .map above
```

Comment:
Since your `deleteBook` api request calls `getBooks` and that call needs the uid to get the books, this could potentially error out without `uid` being passed here.